filebeat/harvester/reader/encode.go

```diff
@@ -33,5 +34,6 @@ func (p Encode) Next() (Message, error) {
 		Ts:      time.Now(),
 		Content: c,
 		Bytes:   sz,
+		Fields:  common.MapStr{},
```

Comment:
The idea is that all reader return an initalised message, so Update fields just work. But the method AddFields below will solve this.